samples/tutorials/mnist/02_Local_Development_with_Docker_Image_Components.ipynb

```diff
@@ -0,0 +1,548 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Copyright 2019 Google Inc. All Rights Reserved.\n",
+    "#\n",
+    "# Licensed under the Apache License, Version 2.0 (the \"License\");\n",
+    "# you may not use this file except in compliance with the License.\n",
+    "# You may obtain a copy of the License at\n",
+    "#\n",
+    "#      http://www.apache.org/licenses/LICENSE-2.0\n",
+    "#\n",
+    "# Unless required by applicable law or agreed to in writing, software\n",
+    "# distributed under the License is distributed on an \"AS IS\" BASIS,\n",
+    "# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n",
+    "# See the License for the specific language governing permissions and\n",
+    "# limitations under the License.\n",
+    "# =============================================================================="
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Local development and docker image components\n",
+    "\n",
+    "- This section assumes that you have already created a program to perform the task required in a particular step of your ML workflow. We will again use the MNIST toy model for demonstration.\n",
+    "\n",
+    "- Then, a Docker container image will be built to package your program. \n",
+    "\n",
+    "- Then, call `kfp.components.ContainerOp` to convert it to a component that can be used in a pipeline. Your component can create outputs that the downstream components can use as inputs\n",
+    "\n",
+    "**Note: Make sure that you have docker installed in the local environment if you want to build the image locally.**"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import kfp\n",
+    "import kfp.gcp as gcp\n",
+    "import kfp.dsl as dsl\n",
+    "import kfp.compiler as compiler\n",
+    "import kfp.components as comp\n",
+    "import datetime\n",
+    "\n",
+    "import kubernetes as k8s"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {
+    "tags": [
+     "parameter"
+    ]
+   },
+   "outputs": [],
+   "source": [
+    "# Required Parameters\n",
+    "PROJECT_ID='<ADD GCP PROJECT HERE>'\n",
+    "GCS_BUCKET='gs://<ADD STORAGE LOCATION HERE>'"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {
+    "tags": [
+     "parameter"
+    ]
+   },
+   "outputs": [],
+   "source": [
+    "# Optional Parameters, but required for running outside kubeflow cluster\n",
+    "HOST = '<ADD HOST NAME TO TALK TO KUBEFLOW PIPELINE HERE>'\n",
+    "CLIENT_ID = '<ADD OAuth CLIENT ID USED BY IAP HERE>'\n",
+    "OTHER_CLIENT_ID = '<ADD OAuth CLIENT ID USED TO OBTAIN AUTH CODES HERE>'\n",
+    "OTHER_CLIENT_SECRET = '<ADD OAuth CLIENT SECRET USED TO OBTAIN AUTH CODES HERE>'"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Create client\n",
```

Comment:
Please apply the same feedback as notebook 01.

Reply:
Yes, moved the content above the block.